src/clp_ffi_py/ir/native/PyDeserializerBuffer.cpp

```diff
@@ -299,9 +299,12 @@ auto PyDeserializerBuffer::metadata_init(PyMetadata* metadata) -> bool {
 }
 
 auto PyDeserializerBuffer::py_getbuffer(Py_buffer* view, int flags) -> int {
-    // Don't need to set the exception message during the failure. The Python level caller will set
-    // the exception and thus overwrite it.
     if (false == is_py_buffer_protocol_enabled()) {
+        PyErr_SetString(
```

Comment:
So is the implication of this PR that we should *always* set the exception even if the caller will overwrite it?

Reply:
No, the comment was wrong. We should raise an exception according to the doc:
https://docs.python.org/3/c-api/typeobj.html#c.PyBufferProcs.bf_getbuffer

(I also fixed the implementation by raising `BufferError` and set `view->obj` to NULL